pennylane/grouping/utils.py

```diff
@@ -658,6 +658,38 @@ def is_qwc(pauli_vec_1, pauli_vec_2):
     return True
 
 
+def are_pauli_words_qwc(lst_pauli_words):
+    """Given a list of observables assumed to be valid Pauli words, determine if they
+     are all mutually qubit-wise commuting.
+
+    This implementation has time complexity ~ O(m * n) for n wires and m Pauli words.
+
+    Args:
+        lst_pauliwords (list[Observable]): list of observables (assumed to be valid Pauli words).
+
+    Returns:
+        (bool): True if they are all qubit-wise commuting, false otherwise.
+    """
+    wire_dict = {}
+
+    for op in lst_pauli_words:  # iterate over the list of  m observables
+        op_names = [op.name] if not isinstance(op.name, list) else op.name
+        op_wires = op.wires.tolist()
+
+        for name, wire in zip(op_names, op_wires):  # iterate over wires, this is at most n
```

Comment:
`name` -> `op_name` renaming could make the code more readable I think.

Reply:
Added!